flaskApi/app.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from zipfile import BadZipFile
+from flask import Flask, request, jsonify
+import app_service
+
+app = Flask(__name__)
+
+# Configure the logger
+logging.basicConfig(level=logging.DEBUG)
+
+
+@app.route('/upload', methods=['POST'])
+def upload_file():
+    # Log request details
+    log_message = f"Request Details: Method={request.method}, URL={request.url}, IP={request.remote_addr}, User-Agent={request.user_agent}"
+    logging.info(log_message)
+
+    # check if file send
+    if 'file' not in request.files:
+        error_msg = 'No file found in request!'
+        logging.error(error_msg)
+        return jsonify({'error_msg': error_msg}), 400
+
+    file = request.files['file']
+
+    # Check if a file is selected
+    if file.filename == '':
+        error_msg = 'No file selected!'
+        logging.error(error_msg)
+        return jsonify({'error_msg': error_msg}), 400
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.